hibernate-core/src/test/java/org/hibernate/test/dialect/functional/PostgresArrayBatchTest.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.dialect.functional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.annotations.BatchSize;
+import org.hibernate.boot.SessionFactoryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.dialect.PostgreSQL81Dialect;
+import org.hibernate.loader.BatchFetchStyle;
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.jdbc.SQLStatementInterceptor;
+import org.hibernate.testing.junit4.BaseNonConfigCoreFunctionalTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+@RequiresDialect(value = { PostgreSQL81Dialect.class })
+@TestForIssue(jiraKey = "HHH-12647")
+public class PostgresArrayBatchTest extends BaseNonConfigCoreFunctionalTestCase {
```

Comment:
Most of the PostgreSQL-specific tests are prefixed with `PostgreSQL`, let's stick to that. I know some of them are not following the convention but let's not add others.

Reply:
I think that's how was named by the original author. I checked the code base and it's a mess. We've been using Postgres, PostgreSql and PostgreSQL.

Reply:
That's what I was saying. Most of them are using `PostgreSQL` and it's in line with the `PostgreSQLDialect`. Let's not make the code more messy than it is.